### PR TITLE
Check a few errors that are currently being ignored

### PR DESCRIPTION
**Pull request by WhyNotHugo**

### CODE REVIEW

File: sbctl.go [hunk 1]
```diff
@@ -41,9 +41,9 @@ func GetESP() (string, error) {
 	}
 
 	for _, location := range espLocations {
-		// "Touch" a file inside all candiadate locations to trigger an
+		// "Read" a file inside all candiadate locations to trigger an
 		// automount if there's an automount partition.
-		os.Stat(fmt.Sprintf("%s/does-not-exist", location))
+		_, _ = os.Stat(fmt.Sprintf("%s/does-not-exist", location))
```
Comment (Foxboron) [inline sbctl.go:46]: Whats the reason to define `_, _ =` in this case? Is this a linter complaint or a personal choice?

File: util.go [hunk 2]
```diff
@@ -172,7 +172,10 @@ func CopyFile(src, dst string) error {
 		return err
 	}
 	defer f.Close()
-	io.Copy(f, source)
+	_, err = io.Copy(f, source)
+	if err != nil {
```
Comment (Foxboron) [inline util.go:176]:
```suggestion
	if _, err = io.Copy(f, source); err != nil {
```

File: sbctl.go [hunk 3]
```diff
@@ -55,7 +55,10 @@ func GetESP() (string, error) {
 	}
 
 	var lsblkRoot LsblkRoot
-	json.Unmarshal(out, &lsblkRoot)
+	err = json.Unmarshal(out, &lsblkRoot)
+	if err != nil {
```
Comment (Foxboron) [inline sbctl.go:59]:
```suggestion
	if err = json.Unmarshal(out, &lsblkRoot); err != nil {
```

File: cmd/sbctl/list-files.go [hunk 4]
```diff
@@ -55,9 +55,9 @@ func RunList(_ *cobra.Command, args []string) error {
 		return err
 	}
 	if cmdOptions.JsonOutput {
-		JsonOut(files)
+		err = JsonOut(files)
```
Comment (Foxboron) [inline cmd/sbctl/list-files.go:58]: Do a return here?

File: sbctl.go (hunk 1 above)
Comment (WhyNotHugo) [reply sbctl.go:46]: Linter complaint: unchecked `err`.

It's almost always a very valid lint, this is a very unusual exception.

File: cmd/sbctl/list-bundles.go [hunk 5]
```diff
@@ -69,9 +69,9 @@ var listBundlesCmd = &cobra.Command{
 			return err
 		}
 		if cmdOptions.JsonOutput {
-			JsonOut(bundles)
+			err = JsonOut(bundles)
```
Comment (Foxboron) [inline cmd/sbctl/list-bundles.go:72]: Return here?

File: bundles.go [hunk 6]
```diff
@@ -33,7 +33,10 @@ func ReadBundleDatabase(dbpath string) (Bundles, error) {
 		return nil, err
 	}
 	bundles := make(Bundles)
-	json.Unmarshal(f, &bundles)
+	err = json.Unmarshal(f, &bundles)
+	if err != nil {
```
Comment (Foxboron) [inline bundles.go:37]:
```suggestion
	if err = json.Unmarshal(f, &bundles); err != nil {
```

File: certs/certs.go [hunk 7]
```diff
@@ -42,7 +42,10 @@ func GetCerts(oem string) (*signature.SignatureDatabase, error) {
 			continue
 		}
 		buf, _ := content.ReadFile(path)
-		sigdb.Append(signature.CERT_X509_GUID, GUID, buf)
+		err := sigdb.Append(signature.CERT_X509_GUID, GUID, buf)
+		if err != nil {
```
Comment (Foxboron) [inline certs/certs.go:46]:
```suggestion
		if err := sigdb.Append(signature.CERT_X509_GUID, GUID, buf); err != nil {
```

File: cmd/sbctl/status.go [hunk 8]
```diff
@@ -88,7 +88,10 @@ func RunStatus(cmd *cobra.Command, args []string) error {
 		stat.Vendors = keys
 	}
 	if cmdOptions.JsonOutput {
-		JsonOut(stat)
+		err := JsonOut(stat)
+		if err != nil {
```
Comment (Foxboron) [inline cmd/sbctl/status.go:92]:
```suggestion
		if err := JsonOut(stat); err != nil {
```

File: database.go [hunk 9]
```diff
@@ -21,7 +21,10 @@ func ReadFileDatabase(dbpath string) (SigningEntries, error) {
 	}
 
 	files := make(SigningEntries)
-	json.Unmarshal(f, &files)
+	err = json.Unmarshal(f, &files)
+	if err != nil {
```
Comment (Foxboron) [inline database.go:25]:
```suggestion
	if err = json.Unmarshal(f, &files); err != nil {
```

**Review by Foxboron — CHANGES_REQUESTED** [review]

**Comment by Foxboron** [comment]
Okay, I didn't bother correcting all the `if err...; err != nil {` stuff in the review :p sorry!

Generally looks okay something we should have been doing for a while :)

File: sbctl.go (hunk 1 above)
Comment (Foxboron) [reply sbctl.go:46]: Ahh! Makes sense :) Then it's fine

**Comment by WhyNotHugo** [comment]
> Generally looks okay something we should have been doing for a while :)

I've been using [`golangci-lint`](https://golangci-lint.run/) to sniff all these out quickly:

```sh
> cat =golint 
#!/bin/sh

podman run --rm -v "$(pwd):/src" -w /src \
  -v "${GOPATH:-$HOME/go}:/go" \
  docker.io/golangci/golangci-lint:v1.46 \
  golangci-lint run --color=always
```

gopls is also very helpful

**Pull request merged**